crates/typst-library/src/math/mod.rs

```diff
@@ -414,7 +415,12 @@ impl LayoutMath for Content {
         }
 
         if let Some(children) = self.to_sequence() {
+            let mut bb = BehavedBuilder::new();
             for child in children {
+                bb.push(child.clone(), ctx.styles());
```

Comment:
The behaved builder has two features: It deals with different kind of elements _and_ it deduplicates styles. Since styling in math is totally half-baked we can just ignore the second feature and pass `StyleChain::default()` everywhere. This removes the lifetime dependence of the builder on the context and thus also makes the `collect` superfluous. So, this could be written as:

```rust
let mut builder = BehavedBuilder::new();
for child in children {
    builder.push(child.clone(), StyleChain::default());
}
for child in builder.elems() {
    child.layout_math(ctx)?;
}
```

This _should_ work okay. I'm not 100% sure whether it breaks in some edge cases, but the same is sadly true for the whole style handling in math. :/

Reply:
One problem might be that `BehavedBuilder::elems` chains `self.staged`. I wonder whether trailing weak spacing is trimmed properly. If not, we could perhaps also use `(child, _) in builder.finish().0.iter()` instead of `child in builder.elems()`.

Reply:
I’m still not satisfied with having to clone each child to push it onto the behaved builder. In fact, there isn’t any reason that `BehavedBuilder` _needs_ to store owned `Content` values instead of references (except that doing the latter would inconvenience `FlowBuilder` and `ParBuilder`).

Reply:
It's not great, but also not that bad since it's all just `Arc` bumps.